dot/state/storage_notify.go

```diff
@@ -60,6 +60,16 @@ func (s *StorageState) RegisterStorageChangeChannel(sub StorageSubscription) (by
 	s.changedLock.Lock()
 	s.subscriptions[id] = &sub
 	s.changedLock.Unlock()
+	// notifyStorageSubscriptions here to send storage value of current state
```

Comment:
@arijitAD I agree this could be done better.  I don't understand your suggestion, I need to learn more about buffered channels.  Are you suggesting replacing the subscription map with a buffered chan of type StorageSubscription?

Reply:
[Fixed-size buffer pool](https://github.com/oxtoacart/bpool/blob/master/bufferpool.go) We can use something like this.
I have created an issue for this. We can fix this later. https://github.com/ChainSafe/gossamer/issues/1477